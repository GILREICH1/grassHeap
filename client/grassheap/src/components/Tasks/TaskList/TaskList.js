import React from 'react';
import TaskItem from '../TaskItem/TaskItem';

function TaskList({tasks}){

  const tasksDisplay = tasks.map((task,i) => <TaskItem key={i} task={task}/>)

  return (
    <div>
    {tasksDisplay}
    </div>
    )
}


export default TaskList;
