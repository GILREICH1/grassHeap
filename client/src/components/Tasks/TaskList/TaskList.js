import React from "react";
import TaskItem from "../TaskItem/TaskItem";

function TaskList({ tasks, deleteThisTask }) {
  const noTasks = (
    <div className="TaskList_empty">
      <p>no tasks for your plants this month</p>
    </div>
  );
  const showTasks = tasks.map((task, i) => (
    <ul key={i}>
      <li>
        <TaskItem deleteThisTask={deleteThisTask} task={task} />
      </li>
    </ul>
  ));

  return (
    <div className="TaskList">{tasks.length === 0 ? noTasks : showTasks}</div>
  );
}

export default TaskList;
