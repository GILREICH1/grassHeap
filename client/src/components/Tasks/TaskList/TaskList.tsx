import TaskItem from '../TaskItem/TaskItem';
import { Task } from '../../../common/types';

interface TaskListProps {
  tasks: Task[];
  deleteThisTask: (_id: string) => void;
}

function TaskList({ tasks, deleteThisTask }: TaskListProps): JSX.Element {
  const noTasks: JSX.Element = (
    <div className="TaskList_empty">
      <p>no tasks for your plants this month</p>
    </div>
  );
  const showTasks: JSX.Element[] = tasks.map((task, i) => (
    <ul key={i}>
      <li>
        <TaskItem deleteThisTask={deleteThisTask} task={task} />
      </li>
    </ul>
  ));

  return (
    <div className="TaskList">{tasks.length === 0 ? noTasks : showTasks}</div>
  );
}

export default TaskList;
