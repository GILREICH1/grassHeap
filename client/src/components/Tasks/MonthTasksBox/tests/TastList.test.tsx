import React from 'react';
import renderer from 'react-test-renderer';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import mocks from './mocks';
import TaskList from '../../TaskList/TaskList';

describe('MonthTasksBox tests', () => {
  let container: Element;
  const mockedProps = {
    deleteThisTask: jest.fn(),
  };

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container);
      container.remove();
    }
  });

  describe('layout', () => {
    test('snapshot without tasks', () => {
      const tree = renderer
        .create(
          <TaskList deleteThisTask={mockedProps.deleteThisTask} tasks={[]} />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('snapshot with tasks', () => {
      const tree = renderer
        .create(
          <TaskList
            deleteThisTask={mockedProps.deleteThisTask}
            tasks={mocks.taskList}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('props', () => {
    test('no tasks', async () => {
      act(() => {
        render(
          <TaskList deleteThisTask={mockedProps.deleteThisTask} tasks={[]} />,
          container,
        );
      });

      expect(container?.textContent).toBe(
        'No tasks for your plants this month!',
      );
    });
  });
});
