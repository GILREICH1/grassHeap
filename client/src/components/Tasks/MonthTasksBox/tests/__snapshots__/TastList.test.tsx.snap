// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthTasksBox tests layout snapshot with tasks 1`] = `
<div
  className="TaskList"
>
  <div
    className="TaskItem"
  >
    <div
      className="div2"
    >
      <h3>
        squash
         
        <img
          className="taskItem_crop_icon"
          onError={[Function]}
          src="https://www.growstuff.org/crops/squash.svg"
        />
      </h3>
    </div>
    <div
      className="div3"
    >
      <p>
        January test task1
      </p>
    </div>
    <div
      className="div1"
    >
      <div
        className="bar"
      />
      <button
        onClick={[Function]}
      >
        X
      </button>
    </div>
  </div>
  <div
    className="TaskItem"
  >
    <div
      className="div2"
    >
      <h3>
        squash
         
        <img
          className="taskItem_crop_icon"
          onError={[Function]}
          src="https://www.growstuff.org/crops/squash.svg"
        />
      </h3>
    </div>
    <div
      className="div3"
    >
      <p>
        October test task2
      </p>
    </div>
    <div
      className="div1"
    />
  </div>
</div>
`;

exports[`MonthTasksBox tests layout snapshot without tasks 1`] = `
<div
  className="TaskList"
>
  <div
    className="TaskList_empty"
  >
    <p>
      No tasks for your plants this month!
    </p>
  </div>
</div>
`;
